language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -349,6 +351,31 @@ private static Hover getFunctionHoverMarkupContent(FunctionSymbol symbol, HoverC
         return hover;
     }
 
+    private static Hover getVariableHoverMarkupContent(VariableSymbol symbol) {
+        Optional<Documentation> documentation = symbol.documentation();
+        if (documentation.isEmpty()) {
+            return getDefaultHoverObject();
+        }
+
+        List<String> hoverContent = new ArrayList<>();
+        if (documentation.get().description().isPresent()) {
+            hoverContent.add(documentation.get().description().get());
+        }
+        TypeSymbol varTypeSymbol = CommonUtil.getRawType(symbol.typeDescriptor());
+        String type = varTypeSymbol.signature();
+        String varName = symbol.getName().isPresent() ? symbol.getName().get() : "";
+        String modifiedVariable = quotedString(type) + " " + varName;
```

Comment:
If variable name is not present, we don't need a space between the varName and the type.